pkg/api/addons.go

```diff
@@ -540,6 +540,13 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		}
 	}
 
+	// Ensure cloud-node-manager is enabled on appropriate upgrades
```

Comment:
I think this will work as-is, but it is also general and may introduce weird stuff in the future. For example, if we introduce additional configuration vectors to this addon w/ defaults, me may (or may not) want to use the default values in an upgrade scenario. In other words, I think the back-compat implementation should be a targeted transformation: "if you were using this specific version of the addon based and are upgrading from this particular version of k8s to this one, do this specific transformation".

What are the specific conditions that this back-compat foo is trying to fix?

Reply:
Yes, using the default for cloud-node-controller should always be safe since it has no config attached...at least currently. But we could run afoul of config changes for the CSI addons.

The specific case I need to test for is `UseCloudControllerManager == true` and upgrade version is 1.16 or greater. If those conditions are met and the addon isn't already enabled, we use the default config for the addon (which will also enable it).

Reply:
Got it, I think this should be fine until we evolve this addon, in which case the "isUpgrade" foo will have to change with it (and UT should be used to support the evolution of all these scenarios)